drivers/scheduler/spec/spec.go

```diff
@@ -14,3 +14,18 @@ type AppSpec struct {
 	// Enabled indicates if the application is enabled in the factory
 	Enabled bool
 }
+
+// DeepCopy Creates a copy of the AppSpec
+func (in *AppSpec) DeepCopy() *AppSpec {
+	if in == nil {
+		return nil
+	}
+	out := new(AppSpec)
+	out.Key = in.Key
+	out.Enabled = in.Enabled
+	out.SpecList = make([]interface{}, 0)
+	for _, spec := range in.SpecList {
+		out.SpecList = append(out.SpecList, spec)
```

Comment:
not sure how this will work when spec is a pointer
as we discussed, in future when we run tests in parallel, and modify the specs, it could be a problem

Reply:
Yes, will need to be changed when we start running tests in parallel.

Reply:
Okay; let's put a TODO here and/or create an issue to track it.

Reply:
Created Issue #166